File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/poll_extension/HMSPollLeaderboardSummaryExtension.kt
package live.hms.hmssdk_flutter.poll_extension

import live.hms.video.polls.network.HMSPollLeaderboardSummary

class HMSPollLeaderboardSummaryExtension {
    companion object {
        fun toDictionary(pollLeaderboardSummary: HMSPollLeaderboardSummary?): HashMap<String, Any?>? {
            if (pollLeaderboardSummary == null) {
                return null
            }

            val map = HashMap<String, Any?>()

            map["average_score"] = pollLeaderboardSummary.averageScore
            map["average_time"] = pollLeaderboardSummary.averageTime
            map["responded_peers_count"] = pollLeaderboardSummary.respondedPeersCount
            map["responded_correctly_peers_count"] = pollLeaderboardSummary.respondedCorrectlyPeersCount
            map["total_peers_count"] = pollLeaderboardSummary.totalPeersCount

            return map
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/Models/polls/HMSPollLeaderboardSummaryExtension.swift
//
//  HMSPollLeaderboardSummaryExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 20/02/24.
//

import Foundation
import HMSSDK

class HMSPollLeaderboardSummaryExtension {

    static func toDictionary(hmsPollLeaderboardSummary: HMSPollLeaderboardSummary?) -> [String: Any?]? {

        if let hmsPollLeaderboardSummary {

            var map = [String: Any?]()

            map["average_score"] = hmsPollLeaderboardSummary.averageScore
            map["average_time"] = hmsPollLeaderboardSummary.averageTime
            map["responded_peers_count"] = hmsPollLeaderboardSummary.respondedPeersCount
            map["responded_correctly_peers_count"] = hmsPollLeaderboardSummary.respondedCorrectlyPeersCount
            map["total_peers_count"] = hmsPollLeaderboardSummary.totalPeersCount

            return map

        } else {
            return nil
        }

    }

}
